src/core/common/instance.hpp

```diff
@@ -745,6 +750,19 @@ template <> inline DuaManager &Instance::Get(void)
 }
 #endif
 
+#if (OPENTHREAD_MTD || OPENTHREAD_FTD) && OPENTHREAD_CONFIG_MLR_ENABLE
+template <> inline MlrManager &Instance::Get(void)
+{
+    return mThreadNetif.mMlrManager;
+}
+
+template <> inline MulticastSubscriptionTable &Instance::Get(void)
```

Comment:
It was named `Multicast Listeners Table` but I renamed to `Multicast Subscription Table` after a discussion with @librasungirl . 

We agreed that `Multicast Listeners Table` in the Spec. is part of the BBR functionalities, but here the `MulticastSubscriptionTable` resides on all nodes (FTD and MTD, BBR and non-BBR). To avoid name conflicts with the `MulticastListenerTable` on BBR (will be implemented later), we deliberately used a different name. 

Here lists some differences between `MulticastSubscriptionTable` and `MulticastListenerTable`:

- MulticastSubscriptionTable resides on all nodes, meanwhile MulticastListenerTable only resides on BBR
- MulticastSubscriptionTable (FTD) entries have ChildMasks, meanwhile MulticastListenerTable entries does not. 
- MulticastSubscriptionTable contains local addresses (MTD/FTD) and children addresses (FTD), meanwhile MulticastListenerTable contains all multicast addresses of all nodes. 

Thoughts?

Reply:
Like @simonlingoogle  mentioned, `MulticastSubscriptionTable` lie in any Thread device, aiming to manage the multicast address subscribed by the device itself (plus its children's subscription for a Router), and the registration/reregistration of these addresses.
`MulticastListenerTable` would lie in BBR device (in future PR), aiming to management the multicast address (scope >=4) interested by the whole network,  based on which triggering MLDv2 on backbone link and determining inbounding forwarding.

How about renaming `MulticastSubscriptionTable` as `SubscriptionAndProxyTable`?  Thus avoiding the confusion due to similarity with future `MulticastListenerTable`, besides, `Subscription` generally indicate it is relative to multicast, and `Proxy` indicates the aggregation on the parent for the multicast address subscribed by the children. 

Thoughts? @jwhui @simonlingoogle

Reply:
Another idea is `MulticastListenerReportTable` or `MulticastListenerRegistrationTable` (whichever term Thread Group settles on). This helps indicate that the table is used for the generation of MLR messages.

Reply:
I actually thought of `MulticastListenerRegistrationTable` or `MlrTable` but didn't choose it because `MulticastListenerRegistrationTable` is too similar to `MulticastListenerTable`. 

Another reason I didn't use `MulticastListenerRegistrationTable` is because that we are planing to remove MAs in `Child` and move them to this table. In such case, this table is required even when `MLR=0`, meaning this table is no longer directly related to the `Multicast Listener Registration` feature, but something more common. We even considered `MulticastAddressTable`. 

> How about renaming MulticastSubscriptionTable as SubscriptionAndProxyTable?

What about `SubscriptionTable` since the current implementation has already defined `subscribed` to be either locally subscribed or proxied. 

That being said, I am fine with any name except `MultiastListenerTable`.

Reply:
I will go with `MulticastListenerRegistrationTable` for now.